packages/web/src/pages/upload-page/fields/stream-availability/usdc-purchase-gated/UsdcPurchaseFields.tsx

```diff
@@ -165,16 +165,23 @@ const PreviewField = (props: TrackAvailabilityFieldsProps) => {
 
 const PriceField = (props: PriceFieldProps) => {
   const { disabled, messaging, fieldName } = props
-  const [{ value }, , { setValue: setPrice }] = useField<number>(fieldName)
+  const [{ value }, , { setValue: setPrice }] = useField<number | null>(
+    fieldName
+  )
   const [humanizedValue, setHumanizedValue] = useState(
     value ? decimalIntegerToHumanReadable(value) : null
   )
 
   const handlePriceChange: ChangeEventHandler<HTMLInputElement> = useCallback(
     (e) => {
       const { human, value } = filterDecimalString(e.target.value)
-      setHumanizedValue(human)
-      setPrice(value)
+      if (value === 0) {
```

Comment:
assuming we check for `null` further downstream and populate the field correctly?